src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProviderFields.cs

```diff
@@ -29,7 +29,7 @@ internal sealed class ModelTypeProviderFields : IObjectTypeFields<InputModelProp
         public IReadOnlyList<Parameter> SerializationParameters { get; }
         public int Count => _fields.Count;
 
-        public ModelTypeProviderFields(InputModelType inputModel, TypeFactory typeFactory, ModelTypeMapping? sourceTypeMapping)
+        public ModelTypeProviderFields(InputModelType inputModel, CSharpType type, TypeFactory typeFactory, ModelTypeMapping? sourceTypeMapping)
```

Comment:
could we kind of specify the `type` parameter to make it more clear? Something like `enclosingType`, or `modelType`

Reply:
updated